source/base/ReferenceCounter.ooc

```diff
@@ -1,20 +1,30 @@
 import threading/Thread
 import Synchronized
 
-ReferenceCounter: class extends Synchronized {
+ReferenceCounter: class {
 	_target: Object
 	count: static Int = 0
 	_count: Int = 0
 	_kill := false
-	init: func (target: Object) {
-		super()
+	_lock: Mutex
+	isSafe: Bool {
+		get { !this _lock instanceOf?(MutexUnsafe) }
+		set(value) {
+			if (this isSafe != value) {
+				this _lock free()
```

Comment:
Nothing, look here
```
MutexGlobal: class extends Mutex {
	_globalMutex := static Mutex new(MutexType Safe)
	init: func
	lock: override func {
		This _globalMutex lock()
	}
	unlock: override func {
		This _globalMutex unlock()
	}
	cleanup: static func {
		This _globalMutex free()
	}
}
```
`free` function does nothing with the global instance of the mutex